backend/lib/product_type.rb

```diff
@@ -0,0 +1,54 @@
+class ProductType
```

Comment:
Aqui temos um ponto interessante.

Os métodos: `shipping_label`, `mail_body`, `is_service?` (ou apenas `service?`), `send_discount?`, por terem suas implementações padrões como `'nil` ou `false`, pode nos sugerir que os filhos dessa classe, não devessem ser filhos, pois eles não irão herdar um comportamento padrão, mas sim, forçar que o método exista de qualquer forma.

Poderíamos substituir isso pelo Duck Typing e deixar as coisas menos travadas. O `self.class` no initialize, evidencia bastante isso, inclusive.